src/main/java/core/mate/academy/model/TruckProducer.java

```diff
@@ -0,0 +1,14 @@
+package core.mate.academy.model;
+
+import core.mate.academy.service.MachineProducer;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer {
```

Comment:
This class should be parameterized with `Truck` type. This will ensure that any implementation NOT using `Machine` or its subtype could NOT be created. Correct implementation should look like this: `public class TruckProducer implements MachineProducer<Truck>`.